ext/zpages/test/tracez_data_aggregator_benchmark.cc

```diff
@@ -0,0 +1,380 @@
+{
+protected:
+  void SetUp(const ::benchmark::State& state) override
+  {
+    SetTracerAggregatorPeer();
+    aggregator_peer->StartPeriodicQueryThread();
+  }
+};
+
+//////////////////////////// BENCHMARK DEFINITIONS /////////////////////////////////
+
+//////////////////////// USER NEVER VISITS WEBPAGE //////////////////////////////
+
+/*
+ * Aggregator handing many spans with the same name, who end instantly. This
+ * checks the scenario where there's only one Tracez name and minimal sorting
+ * of latencies is required, as all spans should be sorted in the same bucket
+ * under the same span name.
+ */
+BENCHMARK_DEFINE_F(TracezAggregator, BM_SingleBucketSingleName)(benchmark::State &state)
+{
+  const int num_spans = state.range(0);
+  for (auto _ : state)
+  {
+    // Do not time span creation, as we're only benchmarking aggregation work
+    state.PauseTiming();
+    StartEndSpans(tracer, num_spans);
+    state.ResumeTiming();
+    aggregator_peer->Aggregate();
+  }
+}
+
+/*
+ * Aggregator handing many spans with unique names, who end instantly. This
+ * checks the scenario where there's many Tracez groups but minimal sorting
+ * of latencies is required. Spans are sorted in different groups but always
+ * in the same bucket.
+ */
+BENCHMARK_DEFINE_F(TracezAggregator, BM_SingleBucketManyNames)(benchmark::State &state)
+{
+  const int num_spans = state.range(0);
+  for (auto _ : state)
+  {
+    state.PauseTiming();
+    StartEndSpans(tracer, num_spans, false);
+    state.ResumeTiming();
+    aggregator_peer->Aggregate();
+  }
+}
+
+/*
+ * Aggregator handing many spans with the same name, who may fall under error,
+ * running, and any latency group. This checks the scenario where there's only
+ * one Tracez groups but there also needs to be sorting of spans into their
+ * respective buckets. Spans are in the same group but sorted to different
+ * buckets.
+ */
+BENCHMARK_DEFINE_F(TracezAggregator, BM_ManyBucketsSingleName)(benchmark::State &state)
+{
+  const int num_spans = state.range(0);
+  for (auto _ : state)
+  {
+    state.PauseTiming();
+    auto running_spans = MakeManySpans(tracer, num_spans, false);
+    state.ResumeTiming();
+    aggregator_peer->Aggregate();
+  }
+}
+
+/*
+ * Aggregator handing many spans with unique names, who may fall under error,
+ * running, and any latency group. This checks the scenario where there's many
+ * Tracez groups and there needs to be sorting of spans into their
+ * respective buckets. Spans are in the different groups and buckets, similar to
+ * likely real use cases.
+ */
+BENCHMARK_DEFINE_F(TracezAggregator, BM_ManyBucketsManyNames)(benchmark::State &state)
+{
+  const int num_spans = state.range(0);
+  for (auto _ : state)
+  {
+    state.PauseTiming();
+    auto running_spans = MakeManySpans(tracer, num_spans, true);
+    state.ResumeTiming();
+    aggregator_peer->Aggregate();
+  }
+}
+
+////////////////// SAME BENCHMARKS, BUT USER VISITS WEBPAGE ///////////////////////////
```

Comment:
If those are the exact same benchmarks to avoid replicating the code you can use templated benchmark: https://github.com/google/benchmark#templated-benchmarks  
And then if the boolean is true, call aggregator_peer->StartPeriodicQueryThread() prior to the for loop in each benchmark.